apps/ejabberd/src/mod_amp.erl

```diff
@@ -166,39 +138,39 @@ determine_strategy(Packet, From, Event) ->
     ejabberd_hooks:run_fold(amp_determine_strategy, host(From),
                             amp_strategy:null_strategy(), [From, To, Packet, Event]).
 
--spec fold_apply_rules(exml:element(), jid(), amp_strategy(), [amp_rule()]) ->
-                              no_match | {matched | undecided, amp_rule()}.
-fold_apply_rules(_, _, _, []) -> 'no_match';
-fold_apply_rules(Packet, From, Strategy, [Rule|Rest]) ->
-    case resolve_condition(From, Strategy, Rule) of
-        no_match -> fold_apply_rules(Packet, From, Strategy, Rest);
-        Status -> {Status, Rule}
+apply_rules(F, Rules) ->
+    [Rule#amp_rule{result = F(Rule)} || Rule <- Rules].
+
+-spec resolve_condition(jid(), amp_strategy(), amp_event(), amp_rule()) -> amp_match_result().
+resolve_condition(From, Strategy, Event, Rule) ->
+    Result = ejabberd_hooks:run_fold
```

Comment:
My eye parser raised a warning reading this line. It'd read better if `(` is in this line, and not below.
This may be only my parser's oddity, though.